src/bundler.ts

```diff
@@ -386,12 +387,93 @@ class Bundler {
    * Given a URL to an entry-point html document, produce a single document
    * with HTML imports, external stylesheets and external scripts inlined,
    * according to the options for this Bundler.
+   *
+   * Given Multiple urls, produces a sharded build.
    */
-  async bundle(url: string): Promise<DocumentCollection> {
+  async bundle(bundles: string[], heuristic?: BundleOrchestrator):
+      Promise<DocumentCollection> {
     if (!this.analyzer) {
       throw new Error('No analyzer provided.');
     }
-    const analyzedRoot = await this.analyzer.analyzeRoot(url);
+    const doc = await this._bundleDocument(
+        bundles[0], this.excludes, this.stripExcludes, this.analyzer);
+    const collection = new Map<string, ASTNode>();
+    collection.set(bundles[0], doc);
+    return collection;
+    // const depsIndex = buildDepsIndex(bundles, this.analyzer) const manifest =
+    //     this._bundleManifest() return this._bundleMultiple(url,
+    //     this.analyzer);
+  }
+
+
+
+  private async _bundleMultiple(entrypoints: string[], analyzer: Analyzer):
+      Promise<DocumentCollection> {
+    const bundles: DocumentCollection = new Map();
+    // Map entrypoints to transitive dependencies.
+    const depsIndex = await buildDepsIndex(entrypoints, analyzer);
+
+
+    // Determine root of dependency graph, if applicable.
+    let root: string|null = null;
+    for (const entrypoint of depsIndex.fragmentToDeps.keys()) {
+      let foundRoot = true;
+      for (const dependency of depsIndex.fragmentToDeps.get(entrypoint)!) {
+        if (depsIndex.fragmentToDeps.has(dependency)) {
+          foundRoot = false;
+          break;
+        }
+      }
+      if (!foundRoot) {
+        continue;
+      }
+      if (root) {
+        throw new Error(
+            `entrypoints must form a singly-rooted graph! Two roots found: "${root
+            }" and "${entrypoint}"!`);
+      }
+      root = entrypoint;
+    }
+    if (!root) {
+      throw new Error(
+          'entrypoints must form a singly-rooted graph! No roots found!')
+    }
+
+    for (const entrypointUrl of entrypoints) {
+      let whitelist: Set<string>;
+      if (entrypointUrl == root) {
+        whitelist = depsIndex.fragmentToDeps.get(entrypointUrl)!;
+      } else {
+        whitelist = depsIndex.fragmentToDeps.get(entrypointUrl)!;
+        for (const dependency of whitelist.values()) {
+          console.log('Considering removing: ', dependency);
+          if (dependency in entrypoints) {
+            const dependenciesToExclude =
```

Comment:
I'm confused about building up this exclusion list. Shouldn't we know exactly which dependencies are exclusive to this bundler after building the dependencies indices? We should have one index of (bundle)->[exclusive-dependencies] where [exclusive-dependencies] contains nothing in any other bundle.